controllers/authorController.js

```diff
@@ -1,269 +1,213 @@
-    if (err) {
-      return next(err);
-    }
-    if (author == null) {
-      // No results.
-      var err = new Error("Author not found");
-      err.status = 404;
-      return next(err);
-    }
-    // Success.
-    res.render("author_form", { title: "Update Author", author: author });
-  });
-};
-
-// Handle Author update on POST.
-exports.author_update_post = [
-  // Validate and santize fields.
-  body("first_name")
-    .trim()
-    .isLength({ min: 1 })
-    .escape()
-    .withMessage("First name must be specified.")
-    .isAlphanumeric()
-    .withMessage("First name has non-alphanumeric characters."),
-  body("family_name")
-    .trim()
-    .isLength({ min: 1 })
-    .escape()
-    .withMessage("Family name must be specified.")
-    .isAlphanumeric()
-    .withMessage("Family name has non-alphanumeric characters."),
-  body("date_of_birth", "Invalid date of birth")
-    .optional({ checkFalsy: true })
-    .isISO8601()
-    .toDate(),
-  body("date_of_death", "Invalid date of death")
-    .optional({ checkFalsy: true })
-    .isISO8601()
-    .toDate(),
-
-  // Process request after validation and sanitization.
-  (req, res, next) => {
-    // Extract the validation errors from a request.
-    const errors = validationResult(req);
-
-    // Create Author object with escaped and trimmed data (and the old id!)
-    var author = new Author({
-      first_name: req.body.first_name,
-      family_name: req.body.family_name,
-      date_of_birth: req.body.date_of_birth,
-      date_of_death: req.body.date_of_death,
-      _id: req.params.id,
-    });
-
-    if (!errors.isEmpty()) {
-      // There are errors. Render the form again with sanitized values and error messages.
-      res.render("author_form", {
-        title: "Update Author",
-        author: author,
-        errors: errors.array(),
-      });
-      return;
-    } else {
-      // Data from form is valid. Update the record.
-      Author.findByIdAndUpdate(
-        req.params.id,
-        author,
-        {},
-        function (err, theauthor) {
-          if (err) {
-            return next(err);
-          }
-          // Successful - redirect to genre detail page.
-          res.redirect(theauthor.url);
-        }
-      );
-    }
-  },
-];
+var Author = require("../models/author");
+var async = require("async");
+var Book = require("../models/book");
+
+const { body, validationResult } = require("express-validator");
+const asyncHandler = require("express-async-handler");
+
+// Display list of all Authors.
+exports.author_list = asyncHandler(async (req, res, next) => {
+  const authors = await Author.find().sort({ family_name: 1 });
+  res.render("author_list", {
+    title: "Author List",
+    author_list: authors,
+  });
+});
+
+// Display detail page for a specific Author.
+exports.author_detail = asyncHandler(async (req, res, next) => {
+  // Get details of author and all their books (in parallel)
+  const [author, authors_books] = await Promise.all([
```

Comment:
This is syntax for waiting for multiple promises and returning when all done. Need to explain this as we pulled out he async module.